### PR TITLE
feat(cli): skip generating a default config in prebuild

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- We have a legacy phase in expo prebuild where we generate an app.json if none exists. But this is not needed anymore and it causes some minor issues with upgrading because we inject the sdkVersion and other default metadata.

# How

- Delete step where we write a default app.json if none exists.
- Update the modification function to support writing the attempted modifications to a default location (app.json) if no Expo config exists.
- Delete warning about creating an app.json.

# Test Plan

`rm -rf app.json && npx expo prebuild` -> Should only create an app.json after writing the app identifiers, no extra properties should be included.

### CODE REVIEW

**Review by douglowder — APPROVED** [review]

File: packages/expo-updates/cli/src/utils/modifyConfigAsync.ts [hunk 1]
```diff
@@ -19,21 +19,11 @@ export async function attemptModification(
   }
 }
 
-function logNoConfig() {
```
Comment (wschurman) [inline packages/expo-updates/cli/src/utils/modifyConfigAsync.ts:22]: I think the call hierarchy for this is only during an explicit execution of the expo-updates CLI, which should log when the CLI can't do what it needs to do (modify the config). I think we should revert this change?

**Review by wschurman — CHANGES_REQUESTED** [review]

File: packages/expo-updates/cli/src/utils/modifyConfigAsync.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo-updates/cli/src/utils/modifyConfigAsync.ts:22]: I updated the `modifyConfigAsync` function so it writes a config file when none exists, the only time we can't modify the config now is when it's a dynamic file (js/ts). We previously couldn't do this because parts of global CLI required a physical config file, but that's fully phased out now.

Comment (wschurman) [reply packages/expo-updates/cli/src/utils/modifyConfigAsync.ts:22]: Ah, got it.

**Review by wschurman — APPROVED** [review]
